### PR TITLE
Add mobile language selector

### PR DESCRIPTION
**Pull request by teesloane**
[ticket](https://circleci.atlassian.net/browse/DD-271)

[preview link](http://circleci-doc-preview.s3-website-us-east-1.amazonaws.com/dd-271-preview/?force-all)

*Do not merge until #6097 has merged*

-----

This PR adds a mobile language picker that mimics what we have on outer. There are a few things to note.

This refactors the work done in #6084 so that the language selector code is:

a) in it's own includes file
b) has it's styling handled in its own fille.

There are some unpleasant hackiness going on in order to emulate what we have on outer (which is done with a bit of nicer styling via tailwind and a different mobile menu) so I did my best to think of how we could do this with just html/css and using as little javascript to re-arrange items.

aside: This branch originally branches off of #6084.

#### Before:

image

#### After:

image

image

### CODE REVIEW

File: src/styles/_langselector.scss
Comment (ccgeller) [inline src/styles/_langselector.scss:6]: nice color declarations

File: src/styles/_langselector.scss
Comment (ccgeller) [inline src/styles/_langselector.scss:139]: nit extra space

**Review by ccgeller — CHANGES_REQUESTED** [review]
I am having the issue if my phone is not long enough the green button 'Go to App' or 'Sign up' doesn't show up. It needs a screen size 1200px + when typical phone is 1080 and older phones 680px.
Screen Shot 2022-01-04 at 4 47 50 PM


Screen Shot 2022-01-04 at 4 45 10 PM

**Comment by teesloane** [comment]
> I am having the issue if my phone is not long enough the green button 'Go to App' or 'Sign up' doesn't show up. It needs a screen size 1200px + when typical phone is 1080 and older phones 680px. Screen Shot 2022-01-04 at 4 47 50 PM
> 
> Screen Shot 2022-01-04 at 4 45 10 PM

Thanks for investigating this. I believe what you are reporting is the default of what's on prod right now. Can you test this on prod?

If I'm understanding correctly - the sign up button is not stickied to the bottom of the mobile menu on prod/or in this PR - it going out of view is a default of the mobile menu drawers opening to indicate what page the user is on?

**Review by rsylvian — CHANGES_REQUESTED** [review]
Tested the preview link and I see a couple UI/UX issues (not all your fault but let's address them regardless):

1. It's missing a separator between `Select Language` and `circleci.com`
2. `Select Language` font is too big and not bold enough
3. `Select Language` and `circleci.com` should be aligned to the separator instead of the items in the menu

yours:
Screen Shot 2022-01-04 at 4 04 45 PM

design:
Screen Shot 2022-01-04 at 4 07 58 PM

----

Screen Shot 2022-01-04 at 4 12 34 PM

This view is also missing a few things:

1. Each row needs to have a green arrow at the end
2. The `Sign Up` button is not present

design:
Screen Shot 2022-01-04 at 4 18 53 PM

**Comment by rsylvian** [comment]
> > I am having the issue if my phone is not long enough the green button 'Go to App' or 'Sign up' doesn't show up. It needs a screen size 1200px + when typical phone is 1080 and older phones 680px. Screen Shot 2022-01-04 at 4 47 50 PM
> > Screen Shot 2022-01-04 at 4 45 10 PM
> 
> Thanks for investigating this. I believe what you are reporting is the default of what's on prod right now. Can you test this on prod?
> 
> If I'm understanding correctly - the sign up button is not stickied to the bottom of the mobile menu on prod/or in this PR - it going out of view is a default of the mobile menu drawers opening to indicate what page the user is on?

Correct, this is also an issue in prod. @ccgeller can you file a ticket in our backlog, please?

**Comment by teesloane** [comment]
Good catches on the design stuff. I'll get right on that.

> 2\. The `Sign Up` button is not present.

We discussed the inclusion of the sign up button [here](https://circleci.slack.com/archives/C021NCSEM44/p1640196159195100) - maybe you can weigh in on it?

**Review by tshen815 — APPROVED** [review]
Looks great! Thanks for all the edits 🎉 

NOTE: we'll merge this after #6097 has been merged to master to handle conflicts

**Review by ccgeller — APPROVED** [review]
Looks good, should we do the PR against 'DD-267/request-new-language' until its merged? 👍✌️💕

**Comment by teesloane** [comment]
> Looks good, should we do the PR against 'DD-267/request-new-language' until its merged? 👍✌️💕

I tried but for some reason it made the PR way bigger :\

File: jekyll/_includes/sidebar.html
Comment (rsylvian) [inline jekyll/_includes/sidebar.html:100]: formating here looks not correct?

File: src/js/services/lang.js
Comment (rsylvian) [inline src/js/services/lang.js:54]: I think it would be better if a selected language had a dedicated CSS class and we just toggle it depending on if it's selected or not. Easier to maintain in the long run

File: src/js/site/nav.js
Comment (rsylvian) [inline src/js/site/nav.js:74]: This is more portable:

```suggestion
    dimmer.css('display', 'inherit');
```

File: src/js/site/nav.js
Comment (rsylvian) [inline src/js/site/nav.js:82]:
```suggestion
    dimmer.hide();
```

File: src/js/site/nav.js
Comment (rsylvian) [inline src/js/site/nav.js:91]:
```suggestion
    dimmer.hide();
```

File: src/js/site/nav.js
Comment (rsylvian) [inline src/js/site/nav.js:102]: Why are we attaching a click event on `body`?

File: src/styles/_utilities.scss
Comment (rsylvian) [inline src/styles/_utilities.scss:10]: Nice one! 👏

**Review by rsylvian — CHANGES_REQUESTED** [review]

File: src/js/site/nav.js
Comment (teesloane) [reply src/js/site/nav.js:74]: `.css()` is a jquery method and not applicable on vanilla js.

File: src/js/site/nav.js
Comment (teesloane) [reply src/js/site/nav.js:82]: Same as above (`.hide()` does not exist on vanilla js)

File: src/js/site/nav.js
Comment (teesloane) [reply src/js/site/nav.js]: There is a bug in prod when clicking on the lang picker glob icon. To recreate:

1. Load the page.
2. click anywhere *but* the globe
3. click on the globe to open the language picker

(It won't open).
The code causing this is [here](https://github.com/circleci/circleci-docs/blob/master/src/js/site/nav.js#L60-L73).

File: src/styles/_utilities.scss
Comment (teesloane) [reply src/styles/_utilities.scss:10]: Just a lil' slice of tailwind ;)

File: src/js/site/nav.js
Comment (rsylvian) [reply src/js/site/nav.js]: @teesloane Thanks for the added context. I'm struggling to understand why this code even exists in the first place?

File: src/js/site/nav.js
Comment (rsylvian) [reply src/js/site/nav.js:74]: we got to stop mixing vanilla and jquery all the time 🙈

**Review by rsylvian — CHANGES_REQUESTED** [review]
I'm having issues with the menu when I click out to close it. It does effectively close it but the UI is in a weird state and I cannot bring the menu back after that:

Kapture 2022-01-10 at 18 30 33

Something else that is pretty minor (and maybe already fixed by Tony's latest PR) but the submit button needs to be right-aligned with the input:

Screen Shot 2022-01-10 at 6 28 19 PM

**Review by tshen815 — CHANGES_REQUESTED** [review]
Preview looks fantastic! 🤩 Bit of a nit here but when the input is active the background should switch to #FFFFFF. Thanks for all your work on this!

Screen Shot 2022-01-11 at 9 22 11 AM
Screen Shot 2022-01-11 at 9 22 44 AM

**Comment by teesloane** [comment]
> Preview looks fantastic! 🤩 Bit of a nit here but when the input is active the background should switch to #FFFFFF. Thanks for all your work on this!
> 
> Screen Shot 2022-01-11 at 9 22 11 AM Screen Shot 2022-01-11 at 9 22 44 AM

fixed!

**Review by tshen815 — APPROVED** [review]
Looks good on my end, thanks for all the edits!

**Comment by ccgeller** [comment]
Looks good for me 🎶😁🦾

**Review by rsylvian — COMMENTED** [review]
Very cool! Found one last thing:

The UI is quite "jumpy" when I submit a new language and I believe the `Thank you for your help` message should be right-aligned:

Kapture 2022-01-12 at 16 23 43

File: src/styles/_global-nav.scss
Comment (rsylvian) [inline src/styles/_global-nav.scss:162]: ✂️

File: src/styles/_utilities.scss
Comment (rsylvian) [inline src/styles/_utilities.scss:5]:
```suggestion
```

**Comment by tshen815** [comment]
New updates looking good with the recent changes! One quick thing, when the language form isnt in mobile view, the styling for the submission message is a bit off. It should be left aligned with the input box. Also, theres no period at the end.

Screen Shot 2022-01-14 at 8 57 15 AM

Thanks for all the edits and merges on this, getting real close!

**Comment by teesloane** [comment]
> New updates looking good with the recent changes! One quick thing, when the language form isnt in mobile view, the styling for the submission message is a bit off. It should be left aligned with the input box. Also, theres no period at the end.
> 
> Screen Shot 2022-01-14 at 8 57 15 AM
> 
> Thanks for all the edits and merges on this, getting real close!

Good catch. I refactored to use flexbox for less pixel pushing. Should be good to go.

**Review by rsylvian — APPROVED** [review]
LGTM ✅  Great work!

**Pull request merged**
